---
title: "MetaIBS - large-scale amplicon-based meta analysis of irritable bowel syndrome"
collection: publications
permalink: /publication/2024-01-23-metaIBS
excerpt: 'Irritable Bowel Syndrome (IBS) is a chronic functional bowel disorder causing abdominal discomfort, as well as transit deregulation with constipation and/or diarrhea. The pathophysiology of IBS is poorly understood and believed to be multifactorial. The role of gut microbiota in IBS has been investigated in several case-control studies, in particular via 16S rRNA amplicon sequencing surveys. These studies, however, have not yet led to a consistent picture of significant changes in gut microbial compositions across health and disease. One key bottleneck is the modest cohort sizes of most individual studies and a high diversity of experimental, bioinformatics, and statistical analysis approaches across studies.Results We address these shortcomings by presenting MetaIBS, an open-access data repository and associated meta-analysis workflow of thirteen 16S rRNA amplicon datasets comprising both fecal matter and sigmoid biopsy samples spanning ∽2,500 IBS and healthy individuals. MetaIBS includes a tailored computational framework that (i) enables coherent de novo processing and taxonomic assignments of the raw 16S rRNA amplicon reads across experimental protocols and sequencing technologies, and (ii) statistical workflows for visualization and analysis at different taxonomic ranks and data granularity. Our statistical meta-analysis shows that popular high-level microbiome summary statistics, including Firmicutes/Bacteroidota ratios or diversity indices, are insufficient for reliable discrimination between IBS patients and healthy controls. Fine-grained multi-method differential abundance and classification analysis, however, can identify sets of differentially abundant taxa that replicate across multiple datasets, including Coprococcus eutactus and Alistipes finegoldii.Conclusions MetaIBS provides a curated and reproducible data and (meta-)analysis resource for amplicon-based IBS research at unprecedented scale. MetaIBS allows assessing the heterogeneity of IBS cohorts across multiple experimental protocols, sample types, and IBS phenotypes. Our framework will likely contribute to more coherent insights into the role of the microbiome in IBS and the discovery of reliable microbial IBS biomarkers for follow-up functional and translational studies.'
date: 2024-01-23
venue: 'BioRxiv'
paperurl: 'https://www.biorxiv.org/content/10.1101/2024.01.22.575775v1'
citation: 'Carcy, S., Ostner, J., Tran, V., Menden, M., and Müller, C. L. (2024). "MetaIBS - large-scale amplicon-based meta analysis of irritable bowel syndrome". <i>bioRxiv</i>, 2024.01.22.575775. doi: 10.1101/2024.01.22.575775'
---
Irritable Bowel Syndrome (IBS) is a chronic functional bowel disorder causing abdominal discomfort, as well as transit deregulation with constipation and/or diarrhea. The pathophysiology of IBS is poorly understood and believed to be multifactorial. The role of gut microbiota in IBS has been investigated in several case-control studies, in particular via 16S rRNA amplicon sequencing surveys. These studies, however, have not yet led to a consistent picture of significant changes in gut microbial compositions across health and disease. One key bottleneck is the modest cohort sizes of most individual studies and a high diversity of experimental, bioinformatics, and statistical analysis approaches across studies.Results We address these shortcomings by presenting MetaIBS, an open-access data repository and associated meta-analysis workflow of thirteen 16S rRNA amplicon datasets comprising both fecal matter and sigmoid biopsy samples spanning ∽2,500 IBS and healthy individuals. MetaIBS includes a tailored computational framework that (i) enables coherent de novo processing and taxonomic assignments of the raw 16S rRNA amplicon reads across experimental protocols and sequencing technologies, and (ii) statistical workflows for visualization and analysis at different taxonomic ranks and data granularity. Our statistical meta-analysis shows that popular high-level microbiome summary statistics, including Firmicutes/Bacteroidota ratios or diversity indices, are insufficient for reliable discrimination between IBS patients and healthy controls. Fine-grained multi-method differential abundance and classification analysis, however, can identify sets of differentially abundant taxa that replicate across multiple datasets, including Coprococcus eutactus and Alistipes finegoldii.Conclusions MetaIBS provides a curated and reproducible data and (meta-)analysis resource for amplicon-based IBS research at unprecedented scale. MetaIBS allows assessing the heterogeneity of IBS cohorts across multiple experimental protocols, sample types, and IBS phenotypes. Our framework will likely contribute to more coherent insights into the role of the microbiome in IBS and the discovery of reliable microbial IBS biomarkers for follow-up functional and translational studies.
[Download paper here](https://www.biorxiv.org/content/10.1101/2024.01.22.575775v1)

Recommended citation: Carcy, S., Ostner, J., Tran, V., Menden, M., and Müller, C. L. (2024). "MetaIBS - large-scale amplicon-based meta analysis of irritable bowel syndrome". <i>bioRxiv</i>, 2024.01.22.575775. doi: 10.1101/2024.01.22.575775
